pyproject.toml

```diff
@@ -22,20 +22,20 @@ requires-python = ">=3.10"
 dynamic = ["version"]
 dependencies = [
     "xarray>=2024.06.0",
-    "kerchunk>=0.2.5",
-    "h5netcdf",
     "numpy>=2.0.0",
-    "ujson",
     "packaging",
     "universal-pathlib",
+    "numcodecs",
```

Comment:
The minimum set of dependencies should look like this, and you could probably even make `universal-pathlib` optional for reading from non-remote files...

Reply:
Actually I re-added `ujson`, because we need it for parsing existing kerchunk-formatted references.

Reply:
It might be nice to have more splits in the optional deps depending on the usecase, kind of like [how Xarray handles it](https://github.com/pydata/xarray/blob/b9780e7a32b701736ebcf33d9cb0b380e92c91d5/pyproject.toml#L52). We could have splits for different readers, writers and local/cloud usecases?

Reply:
I was thinking the same, but kept this PR targeted for now.